src/Utility/TextSplitter.cpp

```diff
@@ -0,0 +1,29 @@
+//
+// Created by dawid on 23.10.2020.
+//
+
+#include "TextSplitter.h"
+
+int TextSplitter::rangeWidth(long long int textLength) {
+    return ceil((double) textLength / (double) numOfThreads);
+}
+
+// It's hard cut the text into substrings of equal size without getting out of range sometimes,
+// so when we get out of range a <-1, -1> is pushed back to indicate that this tuple is wrong and thread should not process it
+
+std::vector<std::tuple<long long int, long long int>>& TextSplitter::splitText(long long int textLength, int patternLength) {
+    int width = rangeWidth(textLength);
+
+    for(int i = 0; i < textLength; i += width) {
+        if((i + patternLength - 1) < textLength && (i + width - 1 + patternLength - 1) < textLength) {
+            ranges.push_back(std::make_tuple(i, i + width - 1));
+        }
+        else if((i + patternLength - 1) < textLength && ((i + width - 1 + patternLength - 1) >= textLength)) {
+            ranges.push_back(std::make_tuple(i, textLength - patternLength));
+        } else if((i + patternLength - 1) >= textLength) {
```

Comment:
I don't like this part with putting the invalid pair. We can make this algorithm produce only valid ranges and throw an error if something is really wrong. I have an idea on how to do this. We need to operate on integers. With a integer division operator you can calculate the len of the chain and then use the reminder from this operation to see how many chains must be 1 window longer